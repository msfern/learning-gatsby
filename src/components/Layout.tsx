import * as React from "react";
import { Link, useStaticQuery, graphql } from "gatsby";
import {
  container,
  heading,
  navLinks,
  navLinkItem,
  navLinkText,
  siteTitle,
} from "./Layout.module.css";

export const PureLayout = ({ pageTitle, data, children }) => {
  return (
    <div className={container} data-testid="hero-title">
      <title>
        {pageTitle} | {data.site.siteMetadata.title}
      </title>
      <header className={siteTitle}>{data.site.siteMetadata.title}</header>
      <p>{data.site.siteMetadata.author}</p>
      <nav>
        <ul className={navLinks}>
          <li className={navLinkItem}>
            <Link to="/" className={navLinkText}>
              Home
            </Link>
          </li>
          <li className={navLinkItem}>
            <Link to="/about" className={navLinkText}>
              About
            </Link>
          </li>
          <li className={navLinkItem}>
            <Link to="/blog" className={navLinkText}>
              Blog
            </Link>
          </li>
        </ul>
      </nav>
      <main>
        <h1 className={heading}>{pageTitle}</h1>
        {children}
      </main>
    </div>
  );
};

const Layout = (props) => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `);

  return <PureLayout {...props} data={data} />;
};

export default Layout;
