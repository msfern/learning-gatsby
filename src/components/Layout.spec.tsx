import React from "react";
import renderer from "react-test-renderer";
import Layout from "./Layout";
import * as Gatsby from "../../__mocks__/gatsby";

describe("Layout", () => {
  it("renders correctly", () => {
    const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");
    useStaticQuery.mockImplementation(() => ({
      site: {
        siteMetadata: {
          title: "My Title",
        },
      },
    }));
    const data = {
      site: {
        siteMetadata: {
          author: "Author",
        },
      },
    };
    const tree = renderer
      .create(
        <Layout pageTitle="Layout Test" data={data}>
          Children
        </Layout>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
