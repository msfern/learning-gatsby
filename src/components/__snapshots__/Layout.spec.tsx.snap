// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="container"
  data-testid="hero-title"
>
  <title>
    Layout Test
     | 
    My Title
  </title>
  <header
    className="siteTitle"
  >
    My Title
  </header>
  <p />
  <nav>
    <ul
      className="navLinks"
    >
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/blog"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      className="heading"
    >
      Layout Test
    </h1>
    Children
  </main>
</div>
`;
